### PR TITLE
[CI] Add GCE variances to Data tests

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

This PR configures BuildKite to run Data release tests on GCE. I excluded the `parquet_metadata_resolution` and `shuffle_data_loader` release tests because more work is required to migrate those tests. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/inference_gce.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+cloud_id: {{env["ANYSCALE_CLOUD_ID"]}}
+region: us-west1
+allowed_azs:
+- us-west1-b
+
+max_workers: 999
+
+head_node_type:
+    name: head_node
+    instance_type: n1-standard-64-nvidia-tesla-t4-1
```
Comment (can-anyscale) [inline release/nightly_tests/dataset/inference_gce.yaml:10]: We actually don't have this type of machine, the list is here: https://github.com/anyscale/product/blob/master/backend/utils/allowed_gpus.yaml

**Comment by can-anyscale** [comment]
If you can scan through the list of failing tests real quick and see if there are any obvious such as a wrong type of machines, that would be great.

If the error is not comprehensible then you can merge the PR.

**Review by can-anyscale — APPROVED** [review]

**Comment by amogkam** [comment]
are the failing tests ok?

**Comment by bveeramani** [comment]
@amogkam yeah. Discussed with @can-anyscale and we're going to merge the PR with the failing tests.

**Comment by can-anyscale** [comment]
Hi yes, please merge even if they fail if the failures are not comprehensible or infra failure related. They are running on GCE stack for the first time so we expect some of them to fail. Merge into master will help the infra team to look at them easier. Thankks

**Comment by amogkam** [comment]
ok can you add a pr description please?

**Pull request merged**